cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/RepositorySourceControlMetadataStore.java

```diff
@@ -58,14 +70,87 @@ private StructuredTable getRepositorySourceControlMetadataTable() {
     return repositorySourceControlMetadataTable;
   }
 
+  /**
+   * Scans the source control metadata records based on the specified request parameters, and
+   * invokes the provided consumer for each matched record.
+   *
+   * @param request  The {@link ScanSourceControlMetadataRequest} object specifying the criteria for
+   *                 scanning the metadata records.
+   * @param type     The type of metadata records to scan. Currently, it is
+   *                 {@code APPLICATIONREFERENCE} by default.
+   * @param consumer The consumer to be invoked for each matched metadata record.
+   * @return The number of metadata records scanned and processed.
+   * @throws IOException If an I/O error occurs while scanning the metadata records.
+   */
+  public int scan(ScanSourceControlMetadataRequest request,
+      String type, Consumer<SourceControlMetadataRecord> consumer) throws IOException {
+    Range.Bound startBound = Range.Bound.INCLUSIVE;
+    final Range.Bound endBound = Range.Bound.INCLUSIVE;
+    Collection<Field<?>> startFields = new ArrayList<>();
+
+    startFields.add(
+        Fields.stringField(StoreDefinition.RepositorySourceControlMetadataStore.NAMESPACE_FIELD,
+            request.getNamespace())
+    );
+    startFields.add(
+        Fields.stringField(StoreDefinition.RepositorySourceControlMetadataStore.TYPE_FIELD,
+            type)
+    );
+
+    if (request.getFilter().getSyncStatus() != null) {
+      startFields.add(
+          Fields.booleanField(StoreDefinition.RepositorySourceControlMetadataStore.IS_SYNCED_FIELD,
+              request.getFilter().getSyncStatus().toString().equals("SYNCED")));
+    }
+
+    Collection<Field<?>> endFields = startFields;
+
+    if (request.getScanAfter() != null) {
+      startBound = Range.Bound.EXCLUSIVE;
+      startFields =
+          request.getSortOn() == SortBy.LAST_SYNCED_DATE ? getLastModifiedRangeFields(request, type)
```

Comment:
why not have one function that takes sorton as parameter

Reply:
Agreed. Done. Thanks for the correction